## References

Where I learned to join CSVs: https://www.geeksforgeeks.org/how-to-merge-multiple-csv-files-into-a-single-pandas-dataframe/

Where I learned to export CSVs: https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.to_csv.html

Where I learned to set an ID for a specific column: https://sparkbyexamples.com/pandas/pandas-set-index-to-column-in-dataframe/

Where I learned to clean data to make sure that they are integers: https://sparkbyexamples.com/pandas/pandas-convert-string-to-integer/

Where I learned to sort values using pandas: https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.sort_values.html

In [1]:
import pandas as pd

all_top_songs = pd.concat(map(pd.read_csv, ['2010.csv', '2011.csv', '2012.csv', '2013.csv', '2014.csv', '2015.csv', '2016.csv', '2017.csv', '2018.csv', '2019.csv']), ignore_index = True)
all_top_songs['id'] = all_top_songs.index
all_top_songs[['id', 'year', 'bpm', 'nrgy', 'dnce', 'dB', 'live', 'val', 'dur', 'acous', 'spch', 'pop']] = all_top_songs[['id', 'year', 'bpm', 'nrgy', 'dnce', 'dB', 'live', 'val', 'dur', 'acous', 'spch', 'pop']].astype(int)
print(all_top_songs)

     Unnamed: 0                                              title  \
0             1                                   Hey, Soul Sister   
1             2                               Love The Way You Lie   
2             3                                            TiK ToK   
3             4                                        Bad Romance   
4             5                               Just the Way You Are   
..          ...                                                ...   
598         599                Find U Again (feat. Camila Cabello)   
599         600      Cross Me (feat. Chance the Rapper & PnB Rock)   
600         601  No Brainer (feat. Justin Bieber, Chance the Ra...   
601         602    Nothing Breaks Like a Heart (feat. Miley Cyrus)   
602         603                                   Kills You Slowly   

               artist        top genre  year  bpm  nrgy  dnce  dB  live  val  \
0               Train       neo mellow  2010   97    89    67  -4     8   80   

## What is the most popular song in the dataset?

The code below shows all songs, sorted by popularity.

In [2]:
all_top_songs.sort_values(by='pop', ascending=False)

Unnamed: 0                    title             artist     top genre  \
572         573                 Memories           Maroon 5           pop   
573         574      Lose You To Love Me       Selena Gomez     dance pop   
574         575        Someone You Loved      Lewis Capaldi           pop   
575         576                 Señorita       Shawn Mendes  canadian pop   
576         577        How Do You Sleep?          Sam Smith           pop   
..          ...                      ...                ...           ...   
50           51                    Hello     Martin Solveig      big room   
362         363         L.A.LOVE (la la)             Fergie     dance pop   
442         443        Million Years Ago              Adele  british soul   
138         139  Blow Me (One Last Kiss)               P!nk     dance pop   
267         268          Not a Bad Thing  Justin Timberlake     dance pop   

     year  bpm  nrgy  dnce  dB  live  val  dur  acous  spch  pop   id  
572  2019   91    32    76  -7     8   57  189     84     5   99  572  
573  2019  102    34    51  -9    21    9  206     58     4   97  573  
574  2019  110    41    50  -6    11   45  182     75     3   96  574  
575  2019  117    54    76  -6     9   75  191      4     3   95  575  
576  2019  111    68    48  -5     8   35  202     15     9   93  576  
..    ...  ...   ...   ...  ..   ...  ...  ...    ...   ...  ...  ...  
50   2010  128    98    67  -3    10   45  191      1     3    0   50  
362  2015  202    39    48  -8    26   27  193      2     9    0  362  
442  2016    0     0     0 -60     0    0  227      0     0    0  442  
138  2012  114    92    60  -3    25   75  256      0     4    0  138  
267  2014   86    73    59  -6    38   46  266     27     6    0  267  

[603 rows x 16 columns]

The most popular song, with a popular score of 99, is "Memories" by Maroon 5.

## What artist has the most songs in the dataset?

The code below shows the total number of songs each artist has in the dataset, from most to least.

In [3]:
all_top_songs['artist'].value_counts()

artist
Katy Perry             17
Justin Bieber          16
Rihanna                15
Maroon 5               15
Lady Gaga              14
                       ..
Iggy Azalea             1
5 Seconds of Summer     1
Michael Jackson         1
Disclosure              1
Daddy Yankee            1
Name: count, Length: 184, dtype: int64

Katy Perry has the most songs in the dataset (17).

## What is the most popular genre of music in the dataset?

The code belows shows the list of genres of all songs in the dataset.

In [4]:
all_top_songs['top genre'].value_counts()

top genre
dance pop                    327
pop                           60
canadian pop                  34
barbadian pop                 15
boy band                      15
electropop                    13
british soul                  11
big room                      10
canadian contemporary r&b      9
neo mellow                     9
art pop                        8
complextro                     6
hip pop                        6
australian dance               6
atl hip hop                    5
australian pop                 5
edm                            5
permanent wave                 4
latin                          4
hip hop                        4
colombian pop                  3
tropical house                 3
brostep                        2
downtempo                      2
electronic trap                2
belgian edm                    2
escape room                    2
folk-pop                       2
electro                        2
indie pop                      2


The most popular genre of music in the dataset is dance pop (327 songs).

## Are there any songs in the dataset more than once?

Yes, 19 songs were mentioned twice, as shown below.

In [5]:
all_top_songs['title'].value_counts().head(19)

title
A Little Party Never Killed Nobody (All We Got)    2
All I Ask                                          2
Kissing Strangers                                  2
Written in the Stars (feat. Eric Turner)           2
The Hills                                          2
Love Yourself                                      2
We Are Never Ever Getting Back Together            2
Sugar                                              2
Say Something                                      2
First Time                                         2
Stitches                                           2
I Like It                                          2
Hello                                              2
Castle Walls (feat. Christina Aguilera)            2
Company                                            2
Runnin' (Lose It All)                              2
Marry You                                          2
Just the Way You Are                               2
Here                                    

The songs that were mentioned twice were "A Little Party Never Killed Anybody (All We Got)", "All I Ask", "Kissing Strangers", "Written in the Stars (feat. Eric Turner)", "The Hills", "Love Yourself", "We Are Never Ever Getting Back Together", "Sugar", "Say Something", "First Time", "Stitches", "I Like It", "Hello", "Castle Walls (feat. Christina Aguilera)", "Company", "Runnin' (Lose It All)", "Marry You", "Just the Way You Are" and "Here".

## Is there a relationship between popularity and danceability?

The code below shows the top 10 most popular songs and 10 least popular songs respectively.

In [6]:
all_top_songs.sort_values(by='pop', ascending=False).head(10)

Unnamed: 0                                              title  \
572         573                                           Memories   
573         574                                Lose You To Love Me   
574         575                                  Someone You Loved   
575         576                                           Señorita   
576         577                                  How Do You Sleep?   
578         579                             Trampoline (with ZAYN)   
577         578  South of the Border (feat. Camila Cabello & Ca...   
580         581                                        Truth Hurts   
581         582         Good as Hell (feat. Ariana Grande) - Remix   
579         580                                            Happier   

            artist     top genre  year  bpm  nrgy  dnce  dB  live  val  dur  \
572       Maroon 5           pop  2019   91    32    76  -7     8   57  189   
573   Selena Gomez     dance pop  2019  102    34    51  -9    21    9  206   
574  Lewis Capaldi           pop  2019  110    41    50  -6    11   45  182   
575   Shawn Mendes  canadian pop  2019  117    54    76  -6     9   75  191   
576      Sam Smith           pop  2019  111    68    48  -5     8   35  202   
578          SHAED    electropop  2019  127    46    62  -6    14   50  184   
577     Ed Sheeran           pop  2019   98    62    86  -6     9   67  204   
580          Lizzo   escape room  2019  158    62    72  -3    12   41  173   
581          Lizzo   escape room  2019   96    89    67  -3    74   48  159   
579     Marshmello       brostep  2019  100    79    69  -3    17   67  214   

     acous  spch  pop   id  
572     84     5   99  572  
573     58     4   97  573  
574     75     3   96  574  
575      4     3   95  575  
576     15     9   93  576  
578     56     3   92  578  
577     15     8   92  577  
580     11    11   90  580  
581     30     6   90  581  
579     19     5   90  579

In [7]:
all_top_songs.sort_values(by='pop').head(10)

Unnamed: 0                          title             artist  \
50           51                          Hello     Martin Solveig   
442         443              Million Years Ago              Adele   
138         139        Blow Me (One Last Kiss)               P!nk   
362         363               L.A.LOVE (la la)             Fergie   
267         268                Not a Bad Thing  Justin Timberlake   
103         104  Muny - Album Version (Edited)        Nicki Minaj   
361         362                      Dangerous    Jennifer Hudson   
441         442               Behind Your Back      Nelly Furtado   
102         103                       Firework         Katy Perry   
101         102   Last Friday Night (T.G.I.F.)         Katy Perry   

          top genre  year  bpm  nrgy  dnce  dB  live  val  dur  acous  spch  \
50         big room  2010  128    98    67  -3    10   45  191      1     3   
442    british soul  2016    0     0     0 -60     0    0  227      0     0   
138       dance pop  2012  114    92    60  -3    25   75  256      0     4   
362       dance pop  2015  202    39    48  -8    26   27  193      2     9   
267       dance pop  2014   86    73    59  -6    38   46  266     27     6   
103       dance pop  2011   98    72    80  -5    19   53  227     56    33   
361       dance pop  2015  109    53    90  -5     8   65  255      0     5   
441  canadian latin  2016   98    47    80 -10     7   69  228     25     4   
102       dance pop  2011  124    83    64  -5    11   65  228     14     5   
101       dance pop  2011  126    81    65  -4    67   72  231      0     4   

     pop   id  
50     0   50  
442    0  442  
138    0  138  
362    0  362  
267    0  267  
103    7  103  
361   18  361  
441   18  441  
102   25  102  
101   27  101

The danceability scores are generally varied as seen in both tables, hence there is no clear relationship between popularity and danceability.

## Does having a higher bpm lead to having a higher energy score?

The code below shows the top 10 songs with the highest bpm and the top 10 songs with the lowest bpm.

In [8]:
all_top_songs.sort_values(by='bpm', ascending=False).head(10)

Unnamed: 0                                              title  \
276         277                                    FourFiveSeconds   
362         363                                   L.A.LOVE (la la)   
200         201                How Ya Doin'? (feat. Missy Elliott)   
462         463                                       The Greatest   
314         315                                      I'll Show You   
248         249      Shot Me Down (feat. Skylar Grey) - Radio Edit   
288         289                                            Animals   
277         278  Love Me Like You Do - From "Fifty Shades Of Grey"   
469         470                              Chained To The Rhythm   
440         441                                      Picky - Remix   

             artist         top genre  year  bpm  nrgy  dnce  dB  live  val  \
276         Rihanna     barbadian pop  2015  206    27    58  -6    13   35   
362          Fergie         dance pop  2015  202    39    48  -8    26   27   
200      Little Mix         dance pop  2013  201    95    36  -3    37   51   
462             Sia  australian dance  2017  192    73    67  -6     6   73   
314   Justin Bieber      canadian pop  2015  192    61    36  -7    18    8   
248    David Guetta         dance pop  2014  192    77    35  -4    12    4   
288        Maroon 5               pop  2015  190    74    28  -6    59   33   
277  Ellie Goulding         dance pop  2015  190    61    26  -7    13   28   
469      Katy Perry         dance pop  2017  190    80    45  -5    20   47   
440    Joey Montana             latin  2016  186    81    70  -3    37   69   

     dur  acous  spch  pop   id  
276  188     88     5   80  276  
362  193      2     9    0  362  
200  211      9    48   50  200  
462  210      1    27   76  462  
314  200      5    10   68  314  
248  191      6     5   61  248  
288  231      0     9   76  288  
277  253     25     5   79  277  
469  238      8    17   72  469  
440  225      9     7   29  440

In [9]:
all_top_songs.sort_values(by='bpm').head(10)

Unnamed: 0                           title              artist  \
442         443               Million Years Ago               Adele   
44           45                     You Lost Me  Christina Aguilera   
87           88                             1+1             Beyoncé   
5             6                            Baby       Justin Bieber   
460         461                         Praying               Kesha   
94           95                   Jar of Hearts     Christina Perri   
220         221                      Let Her Go           Passenger   
260         261  Tee Shirt - Soundtrack Version               Birdy   
183         184            Mirrors - Radio Edit   Justin Timberlake   
109         110        I Knew You Were Trouble.        Taylor Swift   

        top genre  year  bpm  nrgy  dnce  dB  live  val  dur  acous  spch  \
442  british soul  2016    0     0     0 -60     0    0  227      0     0   
44      dance pop  2010   43    39    23  -6    14    7  257     85     4   
87      dance pop  2011   63    38    30  -7     7   26  274     38     3   
5    canadian pop  2010   65    86    73  -5    11   54  214      4    14   
460     dance pop  2017   73    39    58  -7    11   32  230     49     3   
94      dance pop  2011   75    35    35  -6    12    9  247     73     3   
220      folk-pop  2014   75    54    51  -7    10   24  253     39     6   
260    neo mellow  2014   76    34    68  -9     9   56  160     84     3   
183     dance pop  2013   77    60    58  -6    43   65  277     14     5   
109           pop  2012   77    47    62  -7     3   68  220      0     4   

     pop   id  
442    0  442  
44    56   44  
87    60   87  
5     77    5  
460   77  460  
94    50   94  
220   77  220  
260   49  260  
183   60  183  
109   77  109

As we can see, the energy scores are very varied for both tables, hence a higher bpm does not necessarily lead to a higher energy score.

## Which genre has the highest energy score?

The code below lists all genres present in the dataset, with dance pop being the most common genre (327 songs).

In [119]:
all_top_songs['top genre'].value_counts()

top genre
dance pop                    327
pop                           60
canadian pop                  34
barbadian pop                 15
boy band                      15
electropop                    13
british soul                  11
big room                      10
canadian contemporary r&b      9
neo mellow                     9
art pop                        8
complextro                     6
hip pop                        6
australian dance               6
atl hip hop                    5
australian pop                 5
edm                            5
permanent wave                 4
latin                          4
hip hop                        4
colombian pop                  3
tropical house                 3
brostep                        2
downtempo                      2
electronic trap                2
belgian edm                    2
escape room                    2
folk-pop                       2
electro                        2
indie pop                      2


The lines of code below show the average energy scores for each genre, listed in order of the genres shown in the table above.

In [49]:
all_top_songs[all_top_songs['top genre'] == 'dance pop'].nrgy.mean()

72.8256880733945

In [67]:
all_top_songs[all_top_songs['top genre'] == 'pop'].nrgy.mean()

68.3

In [68]:
all_top_songs[all_top_songs['top genre'] == 'canadian pop'].nrgy.mean()

68.11764705882354

In [69]:
all_top_songs[all_top_songs['top genre'] == 'barbadian pop'].nrgy.mean()

68.46666666666667

In [73]:
all_top_songs[all_top_songs['top genre'] == 'boy band'].nrgy.mean()

80.26666666666667

In [74]:
all_top_songs[all_top_songs['top genre'] == 'electropop'].nrgy.mean()

65.23076923076923

In [75]:
all_top_songs[all_top_songs['top genre'] == 'british soul'].nrgy.mean()

51.09090909090909

In [76]:
all_top_songs[all_top_songs['top genre'] == 'big room'].nrgy.mean()

77.2

In [77]:
all_top_songs[all_top_songs['top genre'] == 'canadian contemporary r&b'].nrgy.mean()

64.77777777777777

In [78]:
all_top_songs[all_top_songs['top genre'] == 'neo mellow'].nrgy.mean()

40.666666666666664

In [79]:
all_top_songs[all_top_songs['top genre'] == 'art pop'].nrgy.mean()

60.125

In [80]:
all_top_songs[all_top_songs['top genre'] == 'complextro'].nrgy.mean()

78.5

In [111]:
all_top_songs[all_top_songs['top genre'] == 'hip pop'].nrgy.mean()

60.333333333333336

In [110]:
all_top_songs[all_top_songs['top genre'] == 'australian dance'].nrgy.mean()

72.0

In [112]:
all_top_songs[all_top_songs['top genre'] == 'atl hip hop'].nrgy.mean()

82.2

In [114]:
all_top_songs[all_top_songs['top genre'] == 'australian pop'].nrgy.mean()

74.4

In [115]:
all_top_songs[all_top_songs['top genre'] == 'edm'].nrgy.mean()

61.2

In [116]:
all_top_songs[all_top_songs['top genre'] == 'permanent wave'].nrgy.mean()

65.0

In [117]:
all_top_songs[all_top_songs['top genre'] == 'latin'].nrgy.mean()

80.5

In [118]:
all_top_songs[all_top_songs['top genre'] == 'hip hop'].nrgy.mean()

65.25

In [109]:
all_top_songs[all_top_songs['top genre'] == 'colombian pop'].nrgy.mean()

83.33333333333333

In [108]:
all_top_songs[all_top_songs['top genre'] == 'tropical house'].nrgy.mean()

73.66666666666667

In [107]:
all_top_songs[all_top_songs['top genre'] == 'brostep'].nrgy.mean()

77.5

In [106]:
all_top_songs[all_top_songs['top genre'] == 'downtempo'].nrgy.mean()

57.0

In [105]:
all_top_songs[all_top_songs['top genre'] == 'electronic trap'].nrgy.mean()

76.0

In [104]:
all_top_songs[all_top_songs['top genre'] == 'belgian edm'].nrgy.mean()

67.0

In [103]:
all_top_songs[all_top_songs['top genre'] == 'escape room'].nrgy.mean()

75.5

In [102]:
all_top_songs[all_top_songs['top genre'] == 'folk-pop'].nrgy.mean()

35.5

In [100]:
all_top_songs[all_top_songs['top genre'] == 'electro'].nrgy.mean()

73.5

In [99]:
all_top_songs[all_top_songs['top genre'] == 'indie pop'].nrgy.mean()

87.5

In [98]:
all_top_songs[all_top_songs['top genre'] == 'detroit hip hop'].nrgy.mean()

68.5

In [97]:
all_top_songs[all_top_songs['top genre'] == 'candy pop'].nrgy.mean()

64.77777777777777

In [96]:
all_top_songs[all_top_songs['top genre'] == 'acoustic pop'].nrgy.mean()

61.0

In [95]:
all_top_songs[all_top_songs['top genre'] == 'canadian hip hop'].nrgy.mean()

53.0

In [94]:
all_top_songs[all_top_songs['top genre'] == 'baroque pop'].nrgy.mean()

69.0

In [93]:
all_top_songs[all_top_songs['top genre'] == 'celtic rock'].nrgy.mean()

87.0

In [92]:
all_top_songs[all_top_songs['top genre'] == 'moroccan pop'].nrgy.mean()

76.0

In [91]:
all_top_songs[all_top_songs['top genre'] == 'contemporary country'].nrgy.mean()

59.0

In [90]:
all_top_songs[all_top_songs['top genre'] == 'chicago rap'].nrgy.mean()

69.0

In [89]:
all_top_songs[all_top_songs['top genre'] == 'electro house'].nrgy.mean()

88.0

In [88]:
all_top_songs[all_top_songs['top genre'] == 'australian hip hop'].nrgy.mean()

72.0

In [87]:
all_top_songs[all_top_songs['top genre'] == 'canadian latin'].nrgy.mean()

47.0

In [86]:
all_top_songs[all_top_songs['top genre'] == 'house'].nrgy.mean()

73.0

In [84]:
all_top_songs[all_top_songs['top genre'] == 'danish pop'].nrgy.mean()

47.0

In [85]:
all_top_songs[all_top_songs['top genre'] == 'french indie pop'].nrgy.mean()

89.0

In [83]:
all_top_songs[all_top_songs['top genre'] == 'alaska indie'].nrgy.mean()

34.0

In [82]:
all_top_songs[all_top_songs['top genre'] == 'irish singer-songwriter'].nrgy.mean()

66.0

In [72]:
all_top_songs[all_top_songs['top genre'] == 'hollywood'].nrgy.mean()

82.0

In [71]:
all_top_songs[all_top_songs['top genre'] == 'metropopolis'].nrgy.mean()

72.0

In [70]:
all_top_songs[all_top_songs['top genre'] == 'alternative r&b'].nrgy.mean()

40.0

#### French Indie Pop is the genre with the highest energy score (89.0).

## What percentage of songs may have been recorded live?

18 songs, as seen below, have a live score of more than 50.

In [84]:
all_top_songs.sort_values(by='live', ascending=False).head(18)

Unnamed: 0                                       title  \
400         401                                          NO   
581         582  Good as Hell (feat. Ariana Grande) - Remix   
12           13                 Cooler Than Me - Single Mix   
354         355                           Time of Our Lives   
63           64                                     Monster   
101         102                Last Friday Night (T.G.I.F.)   
175         176   Play Hard (feat. Ne-Yo & Akon) - New Edit   
42           43                                        Hard   
292         293                                       Don't   
559         560               Walk On Water (feat. Beyoncé)   
194         195                         Take Back the Night   
338         339       Prayer in C - Robin Schulz Radio Edit   
17           18                        The Time (Dirty Bit)   
58           59                                 Love On Top   
288         289                                     Animals   
45           46                          Not Myself Tonight   
179         180                                        Move   
1             2                        Love The Way You Lie   
494         495                       Touch (feat. Kid Ink)   

                       artist         top genre  year  bpm  nrgy  dnce  dB  \
400            Meghan Trainor         dance pop  2016   92    80    56  -4   
581                     Lizzo       escape room  2019   96    89    67  -3   
12                Mike Posner         dance pop  2010  130    82    77  -5   
354                   Pitbull         dance pop  2015  124    80    72  -6   
63                 Kanye West       chicago rap  2011  125    69    63  -6   
101                Katy Perry         dance pop  2011  126    81    65  -4   
175              David Guetta         dance pop  2013  130    87    72  -3   
42                    Rihanna     barbadian pop  2010  182    75    31  -4   
292                Ed Sheeran               pop  2015   95    61    81  -7   
559                    Eminem   detroit hip hop  2018   82    44    48 -10   
194         Justin Timberlake         dance pop  2013  107    66    59  -5   
338  Lilly Wood and The Prick  french indie pop  2015  123    89    76  -5   
17        The Black Eyed Peas         dance pop  2010  128    81    82  -8   
58                    Beyoncé         dance pop  2011   94    75    65  -5   
288                  Maroon 5               pop  2015  190    74    28  -6   
45         Christina Aguilera         dance pop  2010  120    81    76  -5   
179                Little Mix         dance pop  2013  121    74    85  -5   
1                      Eminem   detroit hip hop  2010   87    93    75  -5   
494                Little Mix         dance pop  2017  102    75    64  -4   

     live  val  dur  acous  spch  pop   id  
400    74   65  214      1    22   70  400  
581    74   48  159     30     6   90  581  
12     70   63  213     18     5   73   12  
354    69   72  229      9     6   45  354  
63     67   10  379      0    20   73   63  
101    67   72  231      0     4   27  101  
175    66   68  209      4     4   67  175  
42     65   16  251      1    11   57   42  
292    64   85  220      1     7   74  292  
559    64   62  304     81    24   65  559  
194    64   33  353      4    16   54  194  
338    62   78  189      2     3   58  338  
17     60   44  308      7     7   70   17  
58     60   65  267      8     9   76   58  
288    59   33  231      0     9   76  288  
45     56   54  186      0     4   55   45  
179    55   77  224      0     9   63  179  
1      52   64  263     24    23   82    1  
494    49   53  203      4    11   59  494

There are a total of 603 songs in the dataset. Hence, the percentage of songs that were recorded live would be:

18/603 = 0.03

#### 3% of songs may have been recorded live.

## What percentage of songs were acoustic?

44 songs, as seen below, have an acoustic score above 50.

In [92]:
all_top_songs.sort_values('acous', ascending=False).head(44)

Unnamed: 0                                              title  \
431         432                                              Start   
255         256                                   Not About Angels   
186         187                                              Clown   
210         211                                          All of Me   
96           97                                       Turning Page   
436         437                      Beautiful Birds (feat. Birdy)   
52           53                                   Someone Like You   
330         331                                      Mark My Words   
276         277                                    FourFiveSeconds   
398         399                                          All I Ask   
473         474                                          All I Ask   
249         250                                      Say Something   
284         285                                      Night Changes   
44           45                                        You Lost Me   
505         506                                            Free Me   
260         261                     Tee Shirt - Soundtrack Version   
271         272                                      Love Yourself   
364         365                                      Love Yourself   
572         573                                           Memories   
36           37  Bound To You - Burlesque Original Motion Pictu...   
559         560                      Walk On Water (feat. Beyoncé)   
329         330                       What Do You Mean? - Acoustic   
528         529                Perfect Duet (Ed Sheeran & Beyoncé)   
574         575                                  Someone You Loved   
352         353                                            St Jude   
8             9         Empire State of Mind (Part II) Broken Down   
94           95                                      Jar of Hearts   
422         423  Wish That You Were Here - From “Miss Peregrine...   
560         561                                          This Town   
424         425                                             Change   
192         193  Atlas - From “The Hunger Games: Catching Fire”...   
273         274                                  Take Me To Church   
411         412                                    Don't Be A Fool   
279         280                                  What Do You Mean?   
211         212                                       Stay With Me   
499         500                    Lust for Life (with The Weeknd)   
573         574                                Lose You To Love Me   
443         444                                       Shape of You   
592         593                     Girls Like You (feat. Cardi B)   
309         310                                           Trumpets   
538         539                               Youth (feat. Khalid)   
103         104                      Muny - Album Version (Edited)   
578         579                             Trampoline (with ZAYN)   
269         270                               I'm Not The Only One   

                     artist                top genre  year  bpm  nrgy  dnce  \
431             John Legend               neo mellow  2016  110     4    52   
255                   Birdy               neo mellow  2014  116    14    41   
186             Emeli Sandé                dance pop  2013  130    23    45   
210             John Legend               neo mellow  2014  120    26    42   
96         Sleeping At Last             acoustic pop  2011  125    38    30   
436               Passenger                 folk-pop  2016   86    17    39   
52                    Adele             british soul  2011  135    33    56   
330           Justin Bieber             canadian pop  2015  114    26    38   
276                 Rihanna            barbadian pop  2015  206    27    58   
398                   Adele             british soul  2016  142 

There are a total of 603 songs in the dataset, hence the percentage of acoustic songs are:
    
44/603 = 0.07

#### 7% of songs are acoustic.

## Which year’s songs have the highest amount of spoken word in the song?

The code below lists the different years of songs present in the dataset by the average number of spoken words.

In [51]:
all_top_songs[all_top_songs['year'] == 2010].spch.mean()

8.882352941176471

The average amount of spoken words in songs in 2010 was 8.88.

In [52]:
all_top_songs[all_top_songs['year'] == 2011].spch.mean()

9.660377358490566

The average amount of spoken words in songs in 2011 was 9.66.

In [61]:
all_top_songs[all_top_songs['year'] == 2012].spch.mean()

5.8

The average amount of spoken words in songs in 2012 was 5.8.

In [54]:
all_top_songs[all_top_songs['year'] == 2013].spch.mean()

8.309859154929578

The average amount of spoken words in songs in 2013 was 8.31.

In [65]:
all_top_songs[all_top_songs['year'] == 2014].spch.mean()

8.672413793103448

The average amount of spoken words in songs in 2014 was 8.67.

In [56]:
all_top_songs[all_top_songs['year'] == 2015].spch.mean()

7.052631578947368

The average amount of spoken words in songs in 2015 was 7.05.

In [57]:
all_top_songs[all_top_songs['year'] == 2016].spch.mean()

8.3625

The average amount of spoken words in songs in 2016 was 8.36.

In [58]:
all_top_songs[all_top_songs['year'] == 2017].spch.mean()

9.784615384615385

The average amount of spoken words in songs in 2017 was 9.78.

In [66]:
all_top_songs[all_top_songs['year'] == 2018].spch.mean()

8.625

The average amount of spoken words in songs in 2018 was 8.63.

In [60]:
all_top_songs[all_top_songs['year'] == 2019].spch.mean()

8.129032258064516

The average amount of spoken words in songs in 2019 was 8.13.

#### 2017 had the highest amount of spoken words in a song on average, of around 9.78 words.

## Create a single CSV

The code below exports all CSVs into one single CSV.

In [ ]:
all_csvs.to_csv('top_spotify_songs.csv')